### PR TITLE
Added test filter support by folder and tag name

### PR DESCRIPTION
**Pull request by nbhoski**
* Added `selectByFolder` and `selectByTag` test filter options .
*  Added integ test for the same.

### CODE REVIEW

**Comment by mcafaro** [comment]
The results of the integ tests do not show up in the GitHub interface because they are triggered by the completion of a job rather than the a push. The push triggers `build-test-publish`. The completion of `build-test-publish` triggers `pre-integ-test`. The completion of `pre-integ-test` triggers `integ-test-promote`.

You can find all jobs here:

https://dev.azure.com/iat-ci-dev/matlab-azure-devops-extension/_build

It looks like there are currently issues in `integ-test-promote`.

File: tasks/run-matlab-tests/v0/task.json [hunk 1]
```diff
@@ -36,6 +36,22 @@
...
             "required": false,
             "helpMarkDown": "Location of the folder containing source code, relative to the project root folder. The specified folder and its subfolders are added to the top of the MATLAB search path. To generate a code coverage report, MATLAB uses only the source code in the specified folder and its subfolders. You can specify multiple folders using a colon-separated or a semicolon-separated list."
+        },
+        {
+            "name": "selectByFolder",
+            "type": "string",
+            "label": "Test folder",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "Location of the folder containing tests, relative to the project root folder. To generate a test suite, MATLAB uses only the tests in the specified folder and its subfolders. You can specify multiple folders using a colon-separated or a semicolon-separated list."
```
Comment (mw-hrastega) [inline tasks/run-matlab-tests/v0/task.json:46]: We use the following text in CircleCI. We can use the same here.


Location of the folder used to select test suite elements, relative to the project root folder.
To generate a test suite, MATLAB uses only the tests in the specified folder and its subfolders.
You can specify multiple folders using a colon-separated or a semicolon-separated list.

**Review by mw-hrastega — CHANGES_REQUESTED** [review]

**Comment by nbhoski** [comment]
@mcafaro  i guess i need a pemission to check these reports https://dev.azure.com/iat-ci-dev/matlab-azure-devops-extension/_build

File: tasks/run-matlab-tests/v0/task.json (hunk 1 above)
Comment (nbhoski) [reply tasks/run-matlab-tests/v0/task.json:46]: Done.

**Comment by nbhoski** [comment]
All tests are passing now

File: tasks/run-matlab-tests/v0/task.json [hunk 2]
```diff
@@ -36,6 +36,22 @@
...
             "required": false,
             "helpMarkDown": "Location of the folder containing source code, relative to the project root folder. The specified folder and its subfolders are added to the top of the MATLAB search path. To generate a code coverage report, MATLAB uses only the source code in the specified folder and its subfolders. You can specify multiple folders using a colon-separated or a semicolon-separated list."
+        },
+        {
+            "name": "selectByFolder",
+            "type": "string",
+            "label": "Test folder",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "Location of the folder used to select test suite elements, relative to the project root folder.To generate a test suite, MATLAB uses only the tests in the specified folder and its subfolders. You can specify multiple folders using a colon-separated or a semicolon-separated list."
```
Comment (mw-hrastega) [inline tasks/run-matlab-tests/v0/task.json:46]: A whitespace is missing before  "To generate a test suite,".

**Review by mw-hrastega — CHANGES_REQUESTED** [review]

**Review by mw-hrastega — APPROVED** [review]

**Review by mcafaro — APPROVED** [review]
Looks good. Please "Squash and merge", particularly because there are a number of unnecessary commits.

**Pull request merged**